packages/vite/src/node/plugins/resolve.ts

```diff
@@ -802,6 +802,11 @@ function resolveDeepImport(
   targetWeb: boolean,
   options: InternalResolveOptions
 ): string | undefined {
+  // id might contain ?query
+  // e.g. when using `<style src="some-pkg/dist/style.css"></style>` in .vue file
+  // the id will be ./dist/style.css?vue&type=style&index=0&src=xxx&lang.css
+  id = id.replace(/\?.*$/, '')
```

Comment:
Maybe something like `id.split('?')[0]` is much more easier to read 🤔

Reply:
Done ✅